File: CS-Examples/07_DocumentOperation/CheckFileFormat.cs
using System;
using System.Windows.Forms;
using System.Text;
using Spire.Doc;
using Spire.Doc.Documents;
using System.Drawing;
using Spire.Doc.Formatting;
using System.IO;

namespace CheckFileFormat
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
               // Define the input file path
				string input = @"..\..\..\..\..\..\Data\Template.docx";

				// Create a new Document object
				Document doc = new Document();

				// Load the document from the specified input file
				doc.LoadFromFile(input);

				// Get the detected format type of the document
				FileFormat ff = doc.DetectedFormatType;

				// Initialize a string to hold the file format information
				string fileFormat = "The file format is ";

				// Use a switch statement to determine the file format and update the fileFormat string accordingly
				switch (ff)
				{
					case FileFormat.Doc:
						fileFormat += "Microsoft Word 97-2003 document.";
						break;
					case FileFormat.Dot:
						fileFormat += "Microsoft Word 97-2003 template.";
						break;
					case FileFormat.Docx:
						fileFormat += "Office Open XML WordprocessingML Macro-Free Document.";
						break;
					case FileFormat.Docm:
						fileFormat += "Office Open XML WordprocessingML Macro-Enabled Document.";
						break;
					case FileFormat.Dotx:
						fileFormat += "Office Open XML WordprocessingML Macro-Free Template.";
						break;
					case FileFormat.Dotm:
						fileFormat += "Office Open XML WordprocessingML Macro-Enabled Template.";
						break;
					case FileFormat.Rtf:
						fileFormat += "RTF format.";
						break;
					case FileFormat.WordML:
						fileFormat += "Microsoft Word 2003 WordprocessingML format.";
						break;
					case FileFormat.Html:
						fileFormat += "HTML format.";
						break;
					case FileFormat.WordXml:
						fileFormat += "Microsoft Word XML format for Word 2007-2013.";
						break;
					case FileFormat.Odt:
						fileFormat += "OpenDocument Text.";
						break;
					case FileFormat.Ott:
						fileFormat += "OpenDocument Text Template.";
						break;
					case FileFormat.DocPre97:
						fileFormat += "Microsoft Word 6 or Word 95 format.";
						break;
					default:
						fileFormat += "Unknown format.";
						break;
				}

				// Display a message box with the file format information
				MessageBox.Show(fileFormat);

				// Dispose of the Document object to release resources
				doc.Dispose();
        }
    }
}

File: VB-Examples/07_DocumentOperation/CheckFileFormat.vb
Imports System.Text
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports Spire.Doc.Formatting
Imports System.IO

Namespace CheckFileFormat
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Load the input file path into a variable
			Dim input As String = "..\..\..\..\..\..\Data\Template.docx"

			' Create a new Document object
			Dim doc As New Document()

			' Load the document from the input file
			doc.LoadFromFile(input)

			' Get the detected file format of the document
			Dim ff As FileFormat = doc.DetectedFormatType

			' Initialize a variable to store the file format description
			Dim file As String = "The file format is "

			' Check the detected file format and assign the corresponding description
			Select Case ff
				Case FileFormat.Doc
					file &= "Microsoft Word 97-2003 document."
				Case FileFormat.Dot
					file &= "Microsoft Word 97-2003 template."
				Case FileFormat.Docx
					file &= "Office Open XML WordprocessingML Macro-Free Document."
				Case FileFormat.Docm
					file &= "Office Open XML WordprocessingML Macro-Enabled Document."
				Case FileFormat.Dotx
					file &= "Office Open XML WordprocessingML Macro-Free Template."
				Case FileFormat.Dotm
					file &= "Office Open XML WordprocessingML Macro-Enabled Template."
				Case FileFormat.Rtf
					file &= "RTF format."
				Case FileFormat.WordML
					file &= "Microsoft Word 2003 WordprocessingML format."
				Case FileFormat.Html
					file &= "HTML format."
				Case FileFormat.WordXml
					file &= "Microsoft Word xml format for word 2007-2013."
				Case FileFormat.Odt
					file &= "OpenDocument Text."
				Case FileFormat.Ott
					file &= "OpenDocument Text Template."
				Case FileFormat.DocPre97
					file &= "Microsoft Word 6 or Word 95 format."
				Case Else
					file &= "Unknown format."
			End Select

            MessageBox.Show(file)
        End Sub
    End Class
End Namespace
